xrayvision/visibility.py

```diff
@@ -5,10 +5,386 @@
+        """
+
+    @property
+    @abc.abstractmethod
+    def observer_coordinate(self) -> Union[SkyCoord, None]:
+        """
+        Location of the observer.
+        """
+
+    @property
+    @abc.abstractmethod
+    def vis_labels(self) -> Union[Iterable[str], None]:
+        """
+        Labels of each visibility.
+        """
+
+
+class VisibilitiesABC(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def visibilities(self) -> Iterable[apu.Quantity]:
+        """
+        Complex numbers representing the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def u(self) -> Iterable[apu.Quantity]:
+        """
+        u-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def v(self) -> Iterable[apu.Quantity]:
+        """
+        v-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_center(self) -> apu.Quantity[apu.deg]:
+        """
+        The position of the phase center of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def meta(self) -> VisMetaABC:
+        """
+        Metadata.
+        """
+
+    @property
+    @abc.abstractmethod
+    def uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Uncertainties on visibilities values.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitudes of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude_uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitude uncertainty of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phases of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_uncertainty(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phase uncertainty of the visibilities.
+        """
+
+
+class Visibilities(VisibilitiesABC):
+    @apu.quantity_input()
+    def __init__(
+        self,
+        visibilities: apu.Quantity,
+        u: apu.Quantity[1 / apu.arcsec],
+        v: apu.Quantity[1 / apu.arcsec],
+        phase_center: apu.Quantity[apu.deg],
+        uncertainty: Union[apu.Quantity, None] = None,
+        vis_labels: Union[Iterable[str], None] = None,
+        observer_coordinate: Union[SkyCoord, None] = None,
+        energy_range: Union[apu.Quantity[apu.keV], None] = None,
+        time_range: Union[Time, None] = None,
+        meta: Any = dict(),
+        amplitude: Union[apu.Quantity, None] = None,
+        amplitude_uncertainty: Union[apu.Quantity, None] = None,
+        phase: Union[apu.Quantity[apu.arcsec], None] = None,
+        phase_uncertainty: Union[apu.Quantity[apu.arcsec], None] = None,
```

Comment:
I would use `Union[apu.Quantity[apu.deg], None]` for both phase and phase_uncertainty